addon/components/frost-guage.js

```diff
@@ -0,0 +1,180 @@
+import Ember from 'ember'
+const {isPresent} = Ember
+import computed, {readOnly} from 'ember-computed-decorators'
+import {Component} from 'ember-frost-core'
+import {PropTypes} from 'ember-prop-types'
+import layout from '../templates/components/frost-guage'
+
+function getPercent (max, min, value) {
+  let valInt = parseInt(value, 10)
+  let maxInt = parseInt(max, 10)
+  let minInt = parseInt(min, 10)
+  let val = Math.min(Math.max(minInt, valInt), maxInt)
+  return ((val) / max)
+}
+
+export default Component.extend({
+  layout,
+  attributeBindings: ['style'],
+
+  propTypes: {
+    // options
+    elementPrefix: PropTypes.string,
+    min: PropTypes.number,
+    max: PropTypes.number,
+    direction: PropTypes.oneOf(['clockwise', 'counterclockwise']),
+    orientation: PropTypes.number,
+    data: PropTypes.arrayOf(PropTypes.object).isRequired,
+    size: PropTypes.number,
+    text: PropTypes.string
+  },
+
+  getDefaultProps () {
+    const elementPrefix = 'frost-guage'
+
+    return {
+      elementPrefix: elementPrefix,
+      classNames: [elementPrefix],
+      min: 0,
+      max: 100,
+      direction: 'clockwise',
+      // will move scale to propTypes once it's required. Consider it unsupported for now.
+      scale: 0,
+      orientation: 0,
+      negative: '#000000',
+      padding: 10
+    }
+  },
+
+  @readOnly
+  @computed('direction')
+  numericalDirection (direction) {
+    return (direction === 'counterclockwise') ? -1 : 1
+  },
+
+  @readOnly
+  @computed('size')
+  style (size) {
+    return Ember.String.htmlSafe('width: ' + size + 'px; height: ' + size + 'px;')
+  },
+
+  @readOnly
+  @computed('orientation', 'numericalDirection')
+  transform (orientation, numericalDirection) {
+    const d = parseInt(numericalDirection, 10)
+    const s = parseInt(this.get('size'), 10)
+    const o = orientation
+    const c = this.get('center')
+    const rotation = `rotate(${o - 90} ${c} ${c}`
+    const scale = `scale(${numericalDirection}, 1)`
+    let translate = ''
+    if (d === -1) {
+      translate = `translate(${s}, 0)`
+    }
+    return `${translate} ${scale} ${rotation})`
+  },
+
+  @readOnly
+  @computed('size')
+  center (size) {
+    return size / 2
+  },
+
+  @readOnly
+  @computed('size')
+  radius (size) {
+    return (parseInt(size, 10) - this.get('padding') * 2) / 2
```

Comment:
The padding is to compensate for the fact that "size" isn't actually the diameter but rather the square where the circles are. Once some width is added to the circle it will start overflowing through the viewing area.